### PR TITLE
Fix molecular-data-counts service and add unit tests

### PR DESCRIPTION
**Pull request by alisman**
Added unit tests for molecular-profile-sample-counts service.  
Changed name of method to avoid colision with other genomicData stuff.
Fixed issue when multiple studies are queried (multiple profiles of same type need to be merged)

### CODE REVIEW

**Comment by haynescd** [comment]
@alisman can you address the sonar issues? seems to be a lot of unused imports

File: src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml [hunk 1]
```diff
@@ -129,21 +129,20 @@
...
-            SELECT replaceOne(genetic_profile.stable_id, concat(sample_derived.cancer_study_identifier,'_'), '') AS value,
-                genetic_profile.stable_id, 
-                genetic_profile.name AS label, 
-                count(sample_profile.genetic_profile_id) AS count FROM sample_profile 
-            LEFT JOIN sample_derived ON sample_profile.sample_id=sample_derived.internal_id
-            LEFT JOIN genetic_profile on sample_profile.genetic_profile_id = genetic_profile.genetic_profile_id
-            <where>
+        SELECT replaceOne(genetic_profile.stable_id, concat(sample_derived.cancer_study_identifier,'_'), '') AS value,
+        genetic_profile.name AS label,
+        count(sample_profile.genetic_profile_id) AS count FROM sample_profile
```
Comment (haynescd) [inline src/main/resources/org/cbioportal/persistence/mybatisclickhouse/StudyViewMapper.xml:137]:
```suggestion
            count(sample_profile.genetic_profile_id) AS count 
        FROM sample_profile
```

Can you format this

File: src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java [hunk 2]
```diff
@@ -78,10 +80,33 @@ public List<ClinicalDataCount> getClinicalDataCounts(StudyViewFilter studyViewFi
...
+        // different profiles based on the type and choose a label
+        // this code just picks the first label, which assumes that the labels will match
+        // across studies. 
+        List<GenomicDataCount> mergedOutput = new ArrayList<>();
+        for (Map.Entry<String,List<GenomicDataCount>> entry : postsPerType.entrySet()) {
+            var dc = new GenomicDataCount();
+            dc.setValue(entry.getKey());
+            // here just snatch the label of the first profile
+            dc.setLabel(entry.getValue().get(0).getLabel());
+            Integer sum = entry.getValue().stream()
```
Comment (haynescd) [inline src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java:102]: I would just suggest updating the variable names.

mergedOutput, counts, postsPerType

**Review by haynescd — DISMISSED** [review]
just had small suggestions

File: src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java (hunk 2 above)
Comment (alisman) [reply src/main/java/org/cbioportal/persistence/mybatisclickhouse/StudyViewMyBatisRepository.java:102]: how's this

**Pull request merged**